# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/gtest/TestAudioCallbackDriver.cpp
# Commit: 0ae30eae800d
# Full Hash: 0ae30eae800dcad73d997f9f12bb56eb537ef86c
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2020-11-06 04:14:34
# Regressor Bug: 1651745
# File Overlap Count: 1
# Description:
#   Bug 1651745 - Don't buffer more than necessary in AudioInputProcessing. r=padenot
#   
#   When opening a second input track, there will already be some data from its
#   first instantiation in the driver's scratch buffer. If we ignore this data, we
#   end up buffering too much in AudioInputProcessing::Pull and tripping an assert.
# ==============================================================================

diff -r a00ad7266f54 -r 0ae30eae800d dom/media/gtest/TestAudioCallbackDriver.cpp
--- a/dom/media/gtest/TestAudioCallbackDriver.cpp	Thu Nov 05 16:43:24 2020 +0000
+++ b/dom/media/gtest/TestAudioCallbackDriver.cpp	Thu Nov 05 16:44:00 2020 +0000
@@ -26,8 +26,8 @@
   MOCK_METHOD4(NotifyOutputData,
                void(AudioDataValue*, size_t, TrackRate, uint32_t));
   MOCK_METHOD0(NotifyInputStopped, void());
-  MOCK_METHOD4(NotifyInputData,
-               void(const AudioDataValue*, size_t, TrackRate, uint32_t));
+  MOCK_METHOD5(NotifyInputData, void(const AudioDataValue*, size_t, TrackRate,
+                                     uint32_t, uint32_t));
   MOCK_METHOD0(DeviceChanged, void());
   /* OneIteration cannot be mocked because IterationResult is non-memmovable and
    * cannot be passed as a parameter, which GMock does internally. */